libraries/group_execute_provider.rb

```diff
@@ -0,0 +1,28 @@
+class Chef
+  class Provider
+    class GroupExecute < Chef::Provider::Execute
+      provides :group_execute
+
+      def_delegators :secondary_groups
+
+      private
+
+      def command
+        command = new_resource.command
+
+        if new_resource.secondary_groups
+          groups = new_resource.secondary_groups
+          if group
+            # Always do the 'primary' group last in the call chain
+            groups = groups + [group]
```

Comment:
```suggestion
            groups << group
```

This is mostly a style suggestion. `<<` is an append operator and aside from being more common it avoids an array allocation for `[group]`.